docs/program_syntax.md

```diff
@@ -0,0 +1,69 @@
+# Program syntax
+
+Syzkaller uses a special textual representation of programs to log executed programs, test its code and persist programs in the corpus. This page provides a brief description of the corresponding syntax. Some useful information can also be found in the [existing examples](/sys/linux/test) and in the program [deserialization code](/prog/encoding.go).
+
+Together with execution options, such a representation provides everything that syz-executor needs to run a program.
+
+An example of a program:
+```
+r0 = syz_open_dev$loop(&(0x7f00000011c0), 0x0, 0x0)
+r1 = openat$6lowpan_control(0xffffffffffffff9c, &(0x7f00000000c0), 0x2, 0x0)
+ioctl$LOOP_SET_FD(r0, 0x4c00, r1)
+```
+
+Each line of this textual representation describes a particular syscall invokation. The pseudo-formal grammar would be as follows.
+
+```
+line = assignment | call
+assignment = resource " = " call
+call = syscall-name "(" [arg ["," arg]*] ")"  ["(" [call-prop ["," call-prop*] ")"]
+arg = "nil" | "AUTO" | const-arg | resource-arg | result-arg | pointer-arg | string-arg | struct-arg | array-arg | union-arg
+const-arg = "0x" hex-integer
+resource-arg = resource ["/" hex-integer] ["+" hex-integer]
+result-arg = "<" resource "=>" arg
+pointer-arg = "&" pointer-arg-addr ["=ANY"] "=" arg
+pointer-arg-addr = "AUTO" | "(" pointer-addr ["/" region-size] ")"
+string-arg = "'" escaped-string "'" | "\"" escaped-string "\""
+struct-arg =  "{" [arg ["," arg]*] "}"
+array-arg = "[" [arg ["," arg]*] "]"
+union-arg = "@" field-name ["=" arg]
+call-prop = prop-name ": " prop-value
+resource = "r" dec-integer
```

Comment:
I would call this `variable`. Variable is a very familiar and non-confusing concept for any programmer.

Reply:
You're about the `resource` keyword there, right?

Reply:
Yes.